lib/rmt/deduplicator.rb

```diff
@@ -2,14 +2,36 @@
 
 class RMT::Deduplicator
 
-  class MismatchException < RuntimeError
-  end
-
   class HardlinkException < RuntimeError
   end
 
   class << self
 
+    def deduplicate(target_file, force_copy: false, track: true)
+      source_file_path = ::RMT::FileValidator.find_valid_file_by_checksum(
+        target_file.checksum, target_file.checksum_type, deep_verify: false
+      )
+
+      return false if source_file_path.nil?
+
+      make_file_dir(target_file.local_path)
+
+      if RMT::Config.deduplication_by_hardlink? && !force_copy
+        hardlink(source_file_path, target_file.local_path)
+      else
+        copy(source_file_path, target_file.local_path)
+      end
+
+      if track
+        ::DownloadedFile.track_file(checksum: target_file.checksum,
```

Comment:
In fact, I've been consistent with the use of the `::` operator, mainly because of the multiple module levels, though `DownloadedFiles` would never conflict with other names. Greping at the `lib/` dir, it seems the codebase is already not consistent about this style. Do you think it's better to "fix" my changes or to make another PR for style normalization?

Reply:
It's not consistent because not everybody understands how ActiveSupport works :slightly_smiling_face: 
Do whatever is easier, if you don't want to go searching for the lines you've changed in this PR -- I'm fine with fixing everything later.

Reply:
I've changed the files I touched in the last commit.